lib/protobuf/field/message_field.rb

```diff
@@ -9,11 +9,7 @@ class MessageField < BaseField
       #
 
       def acceptable?(val)
-        unless val.is_a?(type_class) || val.respond_to?(:to_hash)
-          fail TypeError, "Expected value of type '#{type_class}' for field #{name}, but got '#{val.class}'"
```

Comment:
it's not a behavior change; we do the same thing in the serialization layer when calling `acceptable?(val)`; if it isn't acceptable we fail with the message there (duplication of behavior)